pkg/kubelet/cm/topologymanager/topology_manager.go

```diff
@@ -233,43 +251,86 @@ func (m *manager) Admit(attrs *lifecycle.PodAdmitAttributes) lifecycle.PodAdmitR
 	klog.Infof("[topologymanager] Topology Admit Handler")
 	pod := attrs.Pod
 
-	for _, container := range append(pod.Spec.InitContainers, pod.Spec.Containers...) {
-		if m.policy.Name() == PolicyNone {
+	// Exception - Policy : none
+	if m.policy.Name() == PolicyNone {
+		for _, container := range append(pod.Spec.InitContainers, pod.Spec.Containers...) {
 			err := m.allocateAlignedResources(pod, &container)
 			if err != nil {
-				return lifecycle.PodAdmitResult{
-					Message: fmt.Sprintf("Allocate failed due to %v, which is unexpected", err),
-					Reason:  "UnexpectedAdmissionError",
-					Admit:   false,
-				}
+				return podAdmitError(unexpectedAdmissionError, err)
 			}
-			continue
 		}
+		return lifecycle.PodAdmitResult{Admit: true}
+	}
+	klog.Infof("[topologymanager] Topology Scope : %v", m.scope)
+	if m.scope == podTopologyScope {
+		return m.podAdmit(pod)
+	}
+
+	return m.containerAdmit(pod)
+}
 
+func (m *manager) containerAdmit(pod *v1.Pod) lifecycle.PodAdmitResult {
+	for _, container := range append(pod.Spec.InitContainers, pod.Spec.Containers...) {
 		result, admit := m.calculateAffinity(pod, &container)
 		if !admit {
-			return lifecycle.PodAdmitResult{
-				Message: "Resources cannot be allocated with Topology locality",
-				Reason:  "TopologyAffinityError",
-				Admit:   false,
-			}
+			return podAdmitError(topologyAffinityError, nil)
 		}
-
-		klog.Infof("[topologymanager] Topology Affinity for (pod: %v container: %v): %v", pod.UID, container.Name, result)
+		klog.Infof("[topologymanager] Topology Affinity for (pod: %v container: %v): %v", format.Pod(pod), container.Name, result)
 		if m.podTopologyHints[string(pod.UID)] == nil {
 			m.podTopologyHints[string(pod.UID)] = make(map[string]TopologyHint)
 		}
 		m.podTopologyHints[string(pod.UID)][container.Name] = result
-
 		err := m.allocateAlignedResources(pod, &container)
 		if err != nil {
-			return lifecycle.PodAdmitResult{
-				Message: fmt.Sprintf("Allocate failed due to %v, which is unexpected", err),
-				Reason:  "UnexpectedAdmissionError",
-				Admit:   false,
-			}
+			return podAdmitError(unexpectedAdmissionError, err)
 		}
 	}
 
 	return lifecycle.PodAdmitResult{Admit: true}
 }
+
+func (m *manager) podAdmit(pod *v1.Pod) lifecycle.PodAdmitResult {
```

Comment:
It would be nice if this function mirrored it's `containerAdmit` counterpart a bit better (even if that means pulling functionality currently abstracted out of `containerAdmit` back into it.

Reply:
Done